### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by lokisk1155**

### CODE REVIEW

File: app/controllers/api/friends_controller.rb [hunk 1]
```diff
@@ -85,5 +54,37 @@ def show
...
+#     end 
 
-end 
\ No newline at end of file
+#     if @userFriends.length > 0 
+#          render `api/friends/index`
+#     else 
+#         render json: { errors: 'No Friends!'}
+#     end 
+#  end
```
Comment (dmallon1) [inline app/controllers/api/friends_controller.rb]: do you want to keep all this commented out code?

File: frontend/src/components/HomePage/Middle/PostFeed.js [hunk 2]
```diff
@@ -88,21 +88,28 @@ function PostFeed() {
...
                   <div className="post-header">
                     <Link to={`/ProfilePage/${post.user_id}`}>
-                      <img className="post-pic" src={simpleUsers[post.user_id].profile_picture ? simpleUsers[post.user_id].profile_picture : profilePic}></img>
+                      <img
+                        className="post-pic"
+                        src={
+                          simpleUsers[post.user_id].profile_picture
+                            ? simpleUsers[post.user_id].profile_picture
+                            : profilePic
+                        }
```
Comment (dmallon1) [inline frontend/src/components/HomePage/Middle/PostFeed.js:97]: Would be nice to keep this consistent with other places
```suggestion
                        src={ simpleUsers[post.user_id].profile_picture || profilePic}
```

File: frontend/src/components/HomePage/Middle/PostFeed.js [hunk 3]
```diff
@@ -114,11 +121,11 @@ function PostFeed() {
                   ) : (
                     <div />
                   )}
-                  {checkPost && targetedPost.user_id === currentUser.id && (
+                  {checkPost && targetedPost.user_id === sessionUser.id && (
                     <Modal onClose={() => setCheckPost(false)}>
                       <CreatePostModal
                         type="update"
-                        currentUser={currentUser}
+                        currentUser={sessionUser}
```
Comment (dmallon1) [inline frontend/src/components/HomePage/Middle/PostFeed.js:128]: You should probably update this prop name too right?
```suggestion
                        sessionUser={sessionUser}
```

File: frontend/src/components/NavBar/NavBar.js [hunk 4]
```diff
@@ -57,6 +43,14 @@ function NavBar() {
     return <Redirect to="/login_page" />;
   }
 
+  if (!simpleUsers[user.id]) {
+    return null;
+  }
+
+  const navbarPic = simpleUsers[user.id].profile_picture
+    ? simpleUsers[user.id].profile_picture
+    : profilePic;
```
Comment (dmallon1) [inline frontend/src/components/NavBar/NavBar.js:52]:
```suggestion
  const navbarPic = simpleUsers[user.id].profile_picture || profilePic;
```

File: frontend/src/components/NavBar/NavBar.js [hunk 5]
```diff
@@ -150,3 +144,20 @@ function NavBar() {
...
+//       !searchRef.current.contains(e.target)
+//     ) {
+//       setToggleSearch(false);
+//     }
+//   };
+//   document.addEventListener("click", checkClickElseWhere);
+//   return () => {
+//     document.removeEventListener("click", checkClickElseWhere);
+//   };
+// }, [toggleSearch]);
```
Comment (dmallon1) [inline frontend/src/components/NavBar/NavBar.js:163]: Should we delete this?

File: frontend/src/store/csrf.js [hunk 6]
```diff
@@ -27,15 +21,10 @@ async function csrfFetch(url, options = {}) {
...
-  // call fetch with the url and the updated options hash
+
   const res = await fetch(url, options);
-  // if the response status code is 400 or above, then throw an error with the
-  // error being the response
   if (res.status >= 400) throw res;
-  // if the response status code is under 400, then return the response to the
-  // next promise chain
-  return res;
+  else return res;
```
Comment (dmallon1) [inline frontend/src/store/csrf.js:27]: You don't need this else and it's cleaner without it
```suggestion
  return res;
```

File: frontend/src/store/profilePage.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+      dispatch(setCurrentProfile(userData.user)),
+      dispatch(receiveFriends(friendsData)),
+    ]);
+  const postRes = await csrfFetch(`/api/posts/${id}`);
+  const postData = await postRes.json();
+  Promise.all([
+    dispatch(receivePosts(postData)),
+    dispatch(setCurrentProfile(userData.user)),
+    dispatch(receiveFriends(friendsData)),
+  ]);
```
Comment (dmallon1) [inline frontend/src/store/profilePage.js:24]: You don't need to use `Promise.all()` unless you're doing something with the promise after, like by using `then`, in this case I would just remove it.

File: app/controllers/api/friends_controller.rb (hunk 1 above)
Comment (lokisk1155) [reply app/controllers/api/friends_controller.rb]: Not really

File: frontend/src/components/HomePage/Middle/PostFeed.js (hunk 3 above)
Comment (lokisk1155) [reply frontend/src/components/HomePage/Middle/PostFeed.js:124]: just a bad prop name I should fix, old code

File: frontend/src/components/NavBar/NavBar.js (hunk 5 above)
Comment (lokisk1155) [reply frontend/src/components/NavBar/NavBar.js:161]: actually want to save this

**Pull request merged**